packages/client/src/store/modules/grants.js

```diff
@@ -212,6 +213,43 @@ export default {
       fetchApi.get(`/api/organizations/${rootGetters['users/selectedAgencyId']}/interested-codes`)
         .then((data) => commit('SET_INTERESTED_CODES', data));
     },
+    async getFollowerForGrant({ rootGetters }, { grantId }) {
+      try {
+        return await fetchApi.get(`/api/organizations/${rootGetters['users/selectedAgencyId']}/grants/${grantId}/follow`);
+      } catch (e) {
+        return null;
+      }
```

Comment:
I'm a bit wary of the error-suppression here; would you mind you elaborating a bit?

Edit: Presumably this is because non-`2xx` status codes from these endpoints are not necessarily indicative of an application error, correct? If so, what would you think of doing something like this in the `fetchApi` methods?
```js
export function get(url) {
  const options = {
    credentials: 'include',
    headers: getDefaultHeaders(),
  };
  return fetch(apiURL(url), options).then((r) => {
    if (r.ok) {
      return r.json();
    }
    const err = new Error(text || r.statusText);
    err.response = r;
    return r
      .text()
      .then(() => Promise.reject(err));
  });
}
```
This would (I think) preserve existing usages, but give functions like `getFollowersForGrant()` the ability to inspect responses for additional information, e.g.
```js
async getFollowerForGrant({ rootGetters }, { grantId }) {
  try {
    return await fetchApi.get(`/api/organizations/${rootGetters['users/selectedAgencyId']}/grants/${grantId}/follow`);
  } catch (e) {
    if (e.response.status === 404) {
      return null;
    }
    throw e;
  }
},
```

Reply:
good question - yes, initially this was because a rejection is expected, and errors wouldn't make sense to be shown in the UI here.  I think its normal if these requests throw (and including the response helps), but maybe the question is how to handle errors that occur outside user actions - can we send to datadog, or other?

Reply:
Datadog RUM is [capturing](https://app.datadoghq.com/rum/error-tracking) any unhandled exceptions and `console.error()` calls (along with [a few other error sources](https://docs.datadoghq.com/real_user_monitoring/browser/collecting_browser_errors/?tab=npm#error-sources)).

In general, I think we should ensure that any error that results from a user's action causes an alert/message being displayed to the user, along with collecting it in rum (if the app can display a message and call `datadogRum.addError()`, then no need to re-`throw`). I'm all for keeping a clean console, but only if it means we're not losing the opportunity to learn that an error exists – a `catch` block should either decide that the error is not an error at all, call `datadogRum.addError()`, or re-`throw` the error (as a last resort).

More specifically for this use-case, I think we should ensure the following:
- `2xx` and `404` status codes are not treated as errors
- Any other `4xx` or `5xx` status code results in a user-facing alert **and** an error should be collected in Datadog RUM.

Reply:
Got it - thanks! Added use of datadog/alert for error handling